frontend/components/doc/TableOfContents.vue

```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
+const props = defineProps<{
+  tocLinks: Array<{
+    id: string;
+    text: string;
+    children: { id: string; text: string }[];
```

Comment:
Are there only two levels of links? Can't it be recursive instead?

Reply:
- i beleive it should not be recursive: 
  1. less straightforward implementation
  2. unpredictable behaviour can happen due to recursion
  3. https://content.nuxtjs.org/api/configuration#toc defines the default depth of the object that i'm working with here which is 2, which i wrote the component for
  
it seems reasonable to adjust this on-the-go in separate issue and redefine toc object + component when needed.